src/CoreBundle/Controller/CatalogueController.php

```diff
@@ -179,7 +179,7 @@ public function listSessions(): JsonResponse
     #[Route('/course-extra-fields', name: 'chamilo_core_catalogue_course_extra_fields', methods: ['GET'])]
     public function getCourseExtraFields(SettingsManager $settingsManager): JsonResponse
     {
-        if ('true' !== $settingsManager->getSetting('course.allow_course_extra_field_in_catalog')) {
+        if (empty($settingsManager->getSetting('catalog.course_catalog_settings'))) {
```

Comment:
Same here it is not 'catalog.course_catalog_settings' which should not be empty but the sub parameter "extra_fields_in_search_form"
So something like :
`$settingsManager->getSetting('catalog.course_catalog_settings')['extra_fields_in_search_form']`